drivers/usb/device/usb_dc_nrfx.c

```diff
@@ -466,14 +466,50 @@ static inline struct usbd_event *usbd_evt_alloc(void)
 	return ev;
 }
 
+static void submit_dc_power_event(enum usbd_periph_state state)
+{
+	struct usbd_event *ev = usbd_evt_alloc();
+
+	if (!ev) {
+		return;
+	}
+
+	ev->evt_type = USBD_EVT_POWER;
+	ev->evt.pwr_evt.state = state;
+
+	usbd_evt_put(ev);
+
+	if (usbd_ctx.attached) {
+		usbd_work_schedule();
+	}
+}
+
+#if CONFIG_USB_NRFX_ATTACHED_EVENT_DELAY
+static void attached_evt_delay_handler(struct k_timer *timer)
+{
+	LOG_DBG("ATTACHED event delay done");
+	submit_dc_power_event(USBD_ATTACHED);
+}
+
+K_TIMER_DEFINE(delay_timer, attached_evt_delay_handler, NULL);
```

Comment:
Is there a reason why it is not `static`?

Reply:
I keep forgetting it. Fixed.